### ISSUE TITLE
Build merging function 

### ISSUE THREAD

**Issue by stefan-riedelseifert** [larshp/upDOWNci#11]
Maybe very useful to have the functionality to merge two variants.

**Comment by larshp** [comment]
Hi,
On which level? I guess first step is being able to compare 2 variants?

**Comment by stefan-riedelseifert** [comment]
Hi Lars,

i often have the task to create a Z….-Variant by merging 2 or more other Variants, for example:

ZEUH_DB = FUNCTIONAL_DB + FUNCTIONAL_DB_ADDITION + PERFORMANCE_DB

I have tested this by merging the 3 xml’s inxml-level and it might be work.

So the only tasl is to merge XML-Content.

Tomorrow I test method/IWFND/CL_SUTIL_ODATA_MAPPER.GET_BUSINESS_DATA

Idea is to put in the xml’s to get structured data.

Best regards, Stefan


Best regards

Stefan Riedel-Seifert

________________________________
Expert ABAP Development Technology | ATP Appl. Techn. Prod. and Logistics
Endress+Hauser InfoServe GmbH+Co. KG | Heinrich-von-Stephan-Str. 8 | 79100 Freiburg | Germany
Phone: +49 761 7059 115 | Mobile: +49 160 3637882 | Fax: +49 761 7059 5 115
stefan.riedel-seifert@infoserve.endress.com | www.infoserve.endress.com


________________________________

Endress+Hauser InfoServe GmbH+Co. KG
Location: Weil am Rhein
Registered: Amtsgericht Freiburg i. Br. HRA 411272
Individual liable partner: Endress+Hauser InfoServe Verwaltungs-GmbH, Weil am Rhein
Registered: Amtsgericht Freiburg i.Br. HRB 412621
Managing director: Pieter de Koning

________________________________

According to the General Data Protection Regulation,
we are obliged to inform you when collecting your personal data.

Wie comply with this information duty with the following Data Protection Statement<https://www.de.endress.com/de/cookies-endress+hauser-website>.

________________________________

Disclaimer:

The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential, proprietary, and/or privileged
material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon, this information by persons or entities
other than the intended recipient is prohibited. If you receive this in error, please contact the sender and delete the material from any computer.
This e-mail does not constitute a contract offer, a contract amendment, or an acceptance of a contract offer unless explicitly and conspicuously designated or stated as such.


Von: Lars Hvam [mailto:notifications@github.com]
Gesendet: Sonntag,3.Juni2018 19:54
An: larshp/upDOWNci <upDOWNci@noreply.github.com>
Cc: Stefan Riedel-Seifert <stefan.riedel-seifert@infoserve.endress.com>; Author <author@noreply.github.com>
Betreff: Re: [larshp/upDOWNci] Build merging function (#11)


Hi,
On which level? I guess first step is being able to compare 2 variants?

**Comment by larshp** [comment]
For the above example I think the variants contains different checks, in this case it is quite easy.

But if 2 checks to be combined contains different configurations for the same check, then the user has to decide which one to take, or to somehow merge it

**Comment by stefan-riedelseifert** [comment]
Yes, you’re right. As a mathematican i prefer the union set of all, as a first approach 😊


Best regards

Stefan Riedel-Seifert

________________________________
Expert ABAP Development Technology | ATP Appl. Techn. Prod. and Logistics
Endress+Hauser InfoServe GmbH+Co. KG | Heinrich-von-Stephan-Str. 8 | 79100 Freiburg | Germany
Phone: +49 761 7059 115 | Mobile: +49 160 3637882 | Fax: +49 761 7059 5 115
stefan.riedel-seifert@infoserve.endress.com | www.infoserve.endress.com


________________________________

Endress+Hauser InfoServe GmbH+Co. KG
Location: Weil am Rhein
Registered: Amtsgericht Freiburg i. Br. HRA 411272
Individual liable partner: Endress+Hauser InfoServe Verwaltungs-GmbH, Weil am Rhein
Registered: Amtsgericht Freiburg i.Br. HRB 412621
Managing director: Pieter de Koning

________________________________

According to the General Data Protection Regulation,
we are obliged to inform you when collecting your personal data.

Wie comply with this information duty with the following Data Protection Statement<https://www.de.endress.com/de/cookies-endress+hauser-website>.

________________________________

Disclaimer:

The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential, proprietary, and/or privileged
material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon, this information by persons or entities
other than the intended recipient is prohibited. If you receive this in error, please contact the sender and delete the material from any computer.
This e-mail does not constitute a contract offer, a contract amendment, or an acceptance of a contract offer unless explicitly and conspicuously designated or stated as such.


Von: Lars Hvam [mailto:notifications@github.com]
Gesendet: Sonntag, 3. Juni 2018 21:03
An: larshp/upDOWNci <upDOWNci@noreply.github.com>
Cc: Stefan Riedel-Seifert <stefan.riedel-seifert@infoserve.endress.com>; Author <author@noreply.github.com>
Betreff: Re: [larshp/upDOWNci] Build merging function (#11)


For the above example I think the variants contains different checks, in this case it is quite easy.

But if 2 checks to be combined contains different configurations for the same check, then the user has to decide which one to take, or to somehow merge it
